apps/emqx_schema_registry/src/emqx_schema_registry.erl

```diff
@@ -333,6 +333,20 @@ async_delete_serdes(Names) ->
 to_bin(A) when is_atom(A) -> atom_to_binary(A);
 to_bin(B) when is_binary(B) -> B.
 
+schema_name_bin_to_atom(Bin) when size(Bin) > 255 ->
+    erlang:throw(
+        iolist_to_binary(
+            io_lib:format(
+                "Name is is too long."
+                " Please provide a shorter name (<= 255 bytes)."
+                " The name that is too long: \"~s\"",
+                [Bin]
+            )
+        )
+    );
```

Comment:
should be a part of the config schema?

Reply:
This does not work when the name is a key. At least, this seems like the same problem as we had here https://github.com/emqx/emqx/pull/10911 and then we decided to make a simple fix until HOCON has support for this. Please correct me if I'm wrong.

Reply:
Maybe we could check for the name length closer to the API boundary, like validating the input in the API handlers before attempting to do any operations.

Reply:
@thalesmg `schema_registry:get_schema/1` is called from several API endpoints in `emqx_schema_registry_http_api`. We could of course do the check in all these places instead but I don't see that this makes things better. I would be happy to be convinced why this would be better. My thinking was to do the check as close as possible to where the error happened to make sure that all call sites where covered.

Reply:
From what I can see, we would need to change only 2 places in `emqx_schema_registry_http_api`: the check in `'/schema_registry'(post, _)` and `param_path_schema_name`.  

The later is the type of the `:name` binding, and adding a validator there would add the check to all `/schema_registry/:name` handlers.  

Then, calls to the internal `get_schema` could assume that input was validated by the upper level.